### PR TITLE
Complete pip-tools setup (backport #5486)

### PR DESCRIPTION
**Pull request by greshilov**
## What do these changes do?
Backport #5486.

## Are there changes in behavior for the user?

No.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by webknjaz** [comment]
@greshilov when it's WIP, it's best to turn the PR into a draft

**Comment by greshilov** [comment]
@webknjaz PTAL

Unfortunately, I had to consider python version to produce valid requirements. The most interesting commit is 5f0637fb939dc94cd009dbb3ea6af651147184ce.

If the presented approach is OK, we probably need some CI job to pin dependencies on different platforms.

**Comment by webknjaz** [comment]
I'd like to have a solution that works with dependabot. Can't we have just a few extra files on the top level?

**Comment by webknjaz** [comment]
Also, this turned into a feature PR. We should first just make it work the same way as on master. And only then, explore remaking things on master.

**Comment by webknjaz** [comment]
https://github.com/dependabot/feedback/issues/658

**Comment by greshilov** [comment]
> I'd like to have a solution that works with dependabot. Can't we have just a few extra files on the top level?

Done.

**Comment by greshilov** [comment]
@webknjaz eventually, differences with the master branch are not huge. The only problem I see is that dependencies have to be pinned using `python3.6`. Maybe we should include a comment on that somewhere in development docs.

File: requirements/doc-spelling.in [hunk 1]
```diff
@@ -0,0 +1,2 @@
+-r doc.txt
+sphinxcontrib-spelling==7.1.0; platform_system!="Windows"  # We only use it in Travis CI
```
Comment (webknjaz) [inline requirements/doc-spelling.in:2]: Lol we switched two CI systems after Travis. But nevermind, the comment not vital for this PR.

**Review by webknjaz — APPROVED** [review]
Thanks for all the effort you've put into this!

**Comment by webknjaz** [comment]
> Maybe we should include a comment on that somewhere in development docs.

Do you think this could cause issues with dependabot?

**Comment by greshilov** [comment]
> > Maybe we should include a comment on that somewhere in development docs.
> 
> Do you think this could cause issues with dependabot?

I don't think so, but we'll see :)

**Pull request merged**